Revert http://svn.apache.org/viewvc?view=revision&revision=r1516556 as it doesn't work when svnserve is started as privileged user.

--- subversion/svnserve/svnserve.c
+++ subversion/svnserve/svnserve.c
@@ -625,9 +625,8 @@
   const char *contents = apr_psprintf(pool, "%" APR_PID_T_FMT "\n",
                                              getpid());
 
-  SVN_ERR(svn_io_remove_file2(filename, TRUE, pool));
   SVN_ERR(svn_io_file_open(&file, filename,
-                           APR_WRITE | APR_CREATE | APR_EXCL,
+                           APR_WRITE | APR_CREATE | APR_TRUNCATE,
                            APR_OS_DEFAULT, pool));
   SVN_ERR(svn_io_file_write_full(file, contents, strlen(contents), NULL,
                                  pool));
